server/comms/rpc/rpctypes.go

```diff
@@ -0,0 +1,98 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package rpc
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Error codes
+const (
+	RPCParseError = iota
+	RPCUnknownMethod
+	RPCInternal
+	RPCQuarantineClient
+	RPCVersionUnsupported
+)
+
+const (
+	JSONRPCVersion = "2.0"
+)
+
+type RPCError struct {
+	Code    int
+	Message string
+}
+
+// Request is the general form of a JSON-RPC request. The type of the Params field varies from one response to the next, so it is implemented as an interface{}.
+type Request struct {
+	// Jsonrpc must be exactly "2.0".
+	Jsonrpc string `json:"jsonrpc"`
+	// The name of the remote procedure being called.
+	Method string `json:"method"`
+	// Params can be any struct or array type, or nil. the type of Params varies
+	// with method.
+	Params json.RawMessage `json:"params"`
+	// ID is a unique identifier the client assigns to this request. If the server
+	// responds the the request, the Response will have the same ID field as
+	// Request. Any integer, string or nil pointer is allowed for ID. A nil
+	// pointer signifies that no response is needed, but may not be valid for all
+	// requests.
+	ID interface{} `json:"id"`
+}
+
+// Response is the general form of a JSON-RPC response.  The type of the
+// Result field varies from one response to the next, so it is implemented as an
+// interface{}. The ID field has to be a pointer to allow for a nil value when
+// empty.
+type Response struct {
+	Jsonrpc string          `json:"jsonrpc"`
+	Result  json.RawMessage `json:"result"`
+	Error   *RPCError       `json:"error"`
+	ID      *interface{}    `json:"id"`
+}
```

Comment:
From the client's perspective these messages cannot be distinguished when received without parsing the message as a request first and if that doesn't work proceed to parse as a response. I think that can be avoided by having these types implement a message interface requiring a `MessageType` function. `Message` type will either return a string or an enum type identifying the message. This way messages can be parsed as a `Message` and based on what `MessageType` returns, parsed accordingly. Thoughts?

Reply:
As per matrix discussion, this is an issue, but is probably better fixed by implementing a wrapper. I'll propose the spec update and fix #33. Good call, @dnldd.